src/main/java/sirius/biz/translations/jdbc/SQLTranslations.java

```diff
@@ -0,0 +1,131 @@
+                                                    field.getName(),
+                                                    lang)
+                            .handle();
+        }
+    }
+
+    @Override
+    public void deleteAllTexts(@Nonnull Mapping field) {
+        try {
+            oma.deleteStatement(SQLTranslation.class)
+               .where(Translation.TRANSLATION_DATA.inner(TranslationData.OWNER), owner.getUniqueName())
+               .where(Translation.TRANSLATION_DATA.inner(TranslationData.FIELD), field.getName())
+               .executeUpdate();
+
+            if (owner instanceof Journaled) {
+                JournalData.addJournalEntry(owner, "Deleted all translated texts for " + field.getName());
+            }
+        } catch (SQLException e) {
+            throw Exceptions.handle()
+                            .to(Mixing.LOG)
+                            .error(e)
+                            .withSystemErrorMessage("Failed to delete translation texts for %s: %s",
+                                                    owner.getUniqueName(),
+                                                    field.getName())
+                            .handle();
+        }
+    }
+
+    @Override
+    protected SQLTranslation findOrCreateTranslation(@Nonnull Mapping field, String lang, String text) {
+        if (!isSupportedLanguage(lang)) {
+            throw new IllegalArgumentException(
+                    "lang must be a language code supported by the system (supportedLanguages)!");
+        }
+
+        Optional<SQLTranslation> translation = fetchTranslation(field, lang);
+
+        if (translation.isPresent()) {
+            return translation.get();
+        } else {
+            SQLTranslation sqlTranslation = new SQLTranslation();
+            sqlTranslation.getTranslationData().setOwner(owner.getUniqueName());
+            sqlTranslation.getTranslationData().setField(field.getName());
+            sqlTranslation.getTranslationData().setLang(lang);
+            return sqlTranslation;
+        }
+    }
+
+    @Override
+    protected Optional<SQLTranslation> fetchTranslation(@Nonnull Mapping field, String lang) {
+        return oma.select(SQLTranslation.class)
+                  .eq(Translation.TRANSLATION_DATA.inner(TranslationData.OWNER), owner.getUniqueName())
+                  .eq(Translation.TRANSLATION_DATA.inner(TranslationData.FIELD), field.getName())
+                  .eq(Translation.TRANSLATION_DATA.inner(TranslationData.LANG), lang)
+                  .first();
+    }
+
+    @Override
+    protected List<SQLTranslation> fetchAllTranslations(@Nonnull Mapping field) {
+        return oma.select(SQLTranslation.class)
+                  .eq(Translation.TRANSLATION_DATA.inner(TranslationData.OWNER), owner.getUniqueName())
+                  .eq(Translation.TRANSLATION_DATA.inner(TranslationData.FIELD), field.getName())
+                  .limit(supportedLanguages.size())
```

Comment:
I'd skip that - this might silently suppress a bug we'd rather want to surface here....